controllers/agents/svc/serviceclass_controller.go

```diff
@@ -95,10 +104,57 @@ func (r *ServiceClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	// then, write all the registered services up to the primaza cluster
-	err = r.CreateRegisteredServices(ctx, &serviceClass, *services)
-	if err != nil {
-		reconcileLog.Error(err, "Failed to write registered services")
-		// fallthrough: we still want to write the service class status field
+	if serviceClass.DeletionTimestamp.IsZero() {
```

Comment:
This conditional block is preventing the control to reach set ownership method. I believe we need to place that function above these checks so as to resolve the errors on service namespace cleanup.

Reply:
I tested this locally, this doesn't have an impact on failing tests.  Which makes sense: if we've hit the deletion case, we've already created the service class, and so we've already set the owner reference on it.

Reply:
Yep, CI fails; check the [CI output](https://github.com/primaza/primaza/actions/runs/4601246839/jobs/8128901937?pr=95) for commit 896c289.

Reply:
Yeah it makes sense. Are you able to replicate the test failure locally? Test are passing locally for me. Created a PR https://github.com/primaza/primaza/pull/101 to use scenario-id instead of static name for service class and increasing timeout to 90s , but still see it failing on ci.